src/ripple/app/misc/ValidatorList.h

```diff
@@ -399,36 +400,35 @@ ValidatorList::onConsensusStart (
         }
     }
 
-    // This quorum guarantees sufficient overlap with the trusted sets of other
-    // nodes using the same set of published lists.
-    std::size_t quorum = keyListings_.size()/2 + 1;
-
-    // Increment the quorum to prevent two unlisted validators using the same
-    // even number of listed validators from forking.
-    if (localPubKey_.size() && ! localKeyListed &&
-            rankedKeys.size () > 1 && keyListings_.size () % 2 != 0)
-        ++quorum;
+    // This minimum quorum of 2f + 1 guarantees safe overlap with the trusted
+    // sets of other nodes using the same set of published lists.
+    // Safety is guaranteed with up to 1/3 listed validators being malicious.
+    std::size_t quorum = calculateMinimumQuorum (keyListings_.size(),
+        localPubKey_.size() && !localKeyListed);
 
     JLOG (j_.debug()) <<
         rankedKeys.size() << "  of " << keyListings_.size() <<
         " listed validators eligible for inclusion in the trusted set";
 
     auto size = rankedKeys.size();
 
-    // Use all eligible keys if there is less than 10 listed validators or
-    // only one trusted list
-    if (size < 10 || publisherLists_.size() == 1)
-    {
-        // Try to raise the quorum toward or above 80% of the trusted set
-        std::size_t const targetQuorum = ValidatorList::calculateQuorum (size);
-        if (targetQuorum > quorum)
-            quorum = targetQuorum;
-    }
-    else
+    // Do not require 80% quorum for less than 10 trusted validators
+    if (size >= 10)
```

Comment:
I'd also suggest changing `if(size >=10)` to `if(rankedKeys.size() > 10)`.  I know you set that in the line above, but those lines move apart in future refactorings, it would be best to make it clear the minimum 10 is relative to the `rankedKeys` size rather than say the listed key size.